Mermaid diagram support across the application

daisuke-awaji opened the pull request:
## Description of Changes

Mermaid diagram support across the application and enhances the meeting minutes feature with diagram generation capabilities.

## Checklist

- [x] Modified relevant documentation
- [x] Verified operation in local environment
- [x] Executed `npm run cdk:test` and if there are snapshot differences, execute `npm run cdk:test:update-snapshot` to update snapshots

## Related Issues

https://github.com/aws-samples/generative-ai-use-cases/issues/1349
https://github.com/aws-samples/generative-ai-use-cases/issues/1330
https://github.com/aws-samples/generative-ai-use-cases/issues/1331

## Screenshot / ChatPage (AFTER)

<img width="1822" height="1184" alt="スクリーンショット 2025-12-06 13 55 18" src="https://github.com/user-attachments/assets/42593c1b-f4ad-4e95-9a2f-b33e2fc0555c" />

<img width="1822" height="1184" alt="スクリーンショット 2025-12-06 13 56 13" src="https://github.com/user-attachments/assets/0df8634f-96dc-416f-ad26-7a31691edae3" />


## Screenshot / Meeting Minutes Page (BEFORE)

<img width="1503" height="764" alt="image" src="https://github.com/user-attachments/assets/79726003-9db5-4293-94d7-2e040f522eaf" />

<img width="1502" height="772" alt="image" src="https://github.com/user-attachments/assets/f46ccf38-2d7c-413c-a9a4-cc0383020117" />


## Screenshot / Meeting Minutes Page (AFTER)

<img width="1822" height="1184" alt="スクリーンショット 2025-12-06 13 48 59" src="https://github.com/user-attachments/assets/11154fbe-9d37-407f-991a-87aa67401dc0" />

<img width="1822" height="1184" alt="スクリーンショット 2025-12-06 13 49 07" src="https://github.com/user-attachments/assets/5d175b94-136e-420d-8f9f-2a61ebd055e9" />

<img width="1822" height="1184" alt="スクリーンショット 2025-12-06 13 49 24" src="https://github.com/user-attachments/assets/1c744c90-baff-47f8-933d-511a23672107" />

<img width="1822" height="1184" alt="スクリーンショット 2025-12-06 13 49 48" src="https://github.com/user-attachments/assets/24288d1f-0717-42e1-af10-dd903c1aef7e" />

<img width="1822" height="1184" alt="スクリーンショット 2025-12-06 13 51 11" src="https://github.com/user-attachments/assets/1809ae75-27cd-45fe-9d27-f54232368883" />


<img width="1822" height="1184" alt="スクリーンショット 2025-12-06 13 51 24" src="https://github.com/user-attachments/assets/e4eefb36-27c0-4c3f-af5f-eca8d2e63cf5" />

<img width="804" height="892" alt="image" src="https://github.com/user-attachments/assets/d74d3555-0996-4f78-b99b-7f5b4c77c1c2" />

daisuke-awaji committed:
feat(markdown): add Mermaid diagram support with code toggle

- Create MermaidWithToggle component for rendering Mermaid diagrams
- Add toggle functionality to switch between diagram and source code view
- Support SVG and PNG export/download options
- Add PreRenderer to skip prose styling for mermaid blocks
- Integrate mermaid rendering into Markdown CodeRenderer
- Re-export MermaidWithToggle for backward compatibility

daisuke-awaji committed:
feat(prompts): add Mermaid diagram support to chat system context

Add instruction for AI to use Mermaid diagrams in code blocks when
explaining processes, relationships, or structures in chat responses.
This enables visual representation of complex concepts.

daisuke-awaji committed:
feat(mermaid): auto-switch to diagram view when code stabilizes

- Change initial view mode from 'diagram' to 'code' during streaming
- Add debounce mechanism (500ms) to detect when code stops changing
- Automatically switch to diagram view once code becomes stable
- Improves UX by preventing incomplete diagram renders during streaming

daisuke-awaji committed:
feat(meetingMinutes): add diagram style option for meeting minutes

Add new "diagram" style to meeting minutes generation feature that
creates visual diagram representations of meeting content. Includes:
- New style_diagram translations for all supported languages
- Updated style selector in MeetingMinutesGeneration component
- Added diagram type to MeetingMinutesStyle union
- Added prompt template for diagram generation

daisuke-awaji committed:
fix: memoize recording state change callbacks to prevent infinite loop

Replace inline arrow functions with useCallback-memoized handlers for
onRecordingStateChange props in MeetingMinutesTranscription and
MeetingMinutesRealtimeTranslation components to prevent re-render loops.

以下のエラーが無限に表示されます。

installHook.js:1 Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render. Error Component Stack
    at MeetingMinutesTranscription (MeetingMinutesTranscription.tsx:48:8)
    at div (<anonymous>)
    at div (<anonymous>)
    at div (<anonymous>)
    at Card (Card.tsx:17:15)
    at div (<anonymous>)
    at div (<anonymous>)
    at div (<anonymous>)
    at MeetingMinutesPage (MeetingMinutesPage.tsx:51:17)
    at App (App.tsx:71:17)
    at AuthWithUserpool (AuthWithUserpool.tsx:20:23)

daisuke-awaji committed:
feat(i18n): add panel view translations for meeting minutes

- Add translation keys for panel selection UI (both_panel,
  generation_panel, transcription_panel) across en, ja, ko, th, vi
- Update MeetingMinutesTranscription component layout with
  improved spacing and flex properties

daisuke-awaji committed:
feat(meeting-minutes): add summary and detail style options

- Add 'summary' and 'detail' styles to MeetingMinutesParams type
- Change default minutes style from 'faq' to 'summary'
- Centralize style type definition in prompts/index.ts
- Reorder style options in dropdown for better UX
- Remove duplicate MeetingMinutesStyle type from hook

daisuke-awaji committed:
feat(meeting-minutes): add diagram type options and settings modal

- Add diagram generation options (flowchart, mindmap, sequence, timeline)
- Implement settings modal with gear icon for configuration
- Add toggle functionality for selecting multiple diagram types
- Add i18n translations for diagram options and settings in en/ja

daisuke-awaji committed:
refactor(web): extract MeetingMinutesSettingsModal into separate component

- Move settings modal UI from MeetingMinutesGeneration to dedicated component
- Remove unused imports (Checkbox, ModalDialog, Select, Switch, Textarea)
- Improve code organization and maintainability

daisuke-awaji committed:
feat(meeting-minutes): refactor audio input controls with unified toggle UI

- Replace separate mic/screen audio toggles with TogglePillButton component
- Add input source selection dropdown with desktop and microphone options
- Add translation keys for input_source, microphone, and option in EN/JA
- Relocate start/stop recording buttons to transcript panel header
- Integrate Help component for improved UI consistency
- Streamline audio control interface for better user experience

daisuke-awaji committed:
feat(ui): improve meeting minutes interface styling and responsiveness

- Lighten placeholder text color from gray-500 to gray-400 for better visual hierarchy
- Add horizontal scroll to tab headers for better mobile experience
- Reduce tab text size to improve spacing and readability
- Ensure tab content containers use full height for proper layout

daisuke-awaji committed:
style: refactor meeting minutes UI layout and icon sizing

- Remove unnecessary flexbox classes from select containers
- Reduce bottom padding in settings panel
- Remove outlined prop from start recording button
- Increase mobile icon sizes from 7x7 to 10x10 for better touch targets
- Simplify div styling by removing redundant flex properties

daisuke-awaji committed:
feat: add collapsible settings panel for mobile responsiveness

- Add mobile detection using useIsMobile hook
- Make settings panel collapsible on mobile devices to save screen space
- Add gear and caret icons to settings header for better UX
- Settings expand/collapse based on screen size (collapsed on mobile, expanded on desktop)
- Improve responsive design for meeting minutes translation interface

daisuke-awaji committed:
feat(meeting-minutes): extract control buttons and add settings panel

- Extract recording control buttons into reusable MeetingMinutesControlButtons component
- Add collapsible settings panel with speaker configuration options
- Improve component organization and mobile responsiveness
- Add screen audio support and speaker label management

daisuke-awaji committed:
feat(meeting-minutes): add whiteboard style option and shorten clear button text

- Add new whiteboard style option for meeting minutes generation
- Simplify "clear_minutes" button text across all language files
- Update UI components to support whiteboard style selection

daisuke-awaji committed:
feat(meeting-minutes): add prompt visibility toggle with i18n support

Add view/hide prompt functionality to meeting minutes settings with translations for English, Japanese, Korean, Vietnamese, and Chinese. Includes system prompt preview capability in the settings modal.

Sugi275 commented:
@daisuke-awaji 
大きく分けて、議事録ページの UI 改善、マーメイドの機能追加の主要な 2 点の実装ありがとうございます！
特に、議事録ページの UI は、とても体験がよくなって素敵だと思います。

# Mermaid の図解について
GenU がデフォルトで提供している議事録スタイルの、「要約」「詳細」「質疑応答」「文字起こし」に Mermaid の出力は無くても良いのではないかなと思いました。この議事録機能を利用する方は、Mermaid の図解よりも、まずはテキストのアウトプットを期待しているためです。他の議事録スタイルの「図解」「ホワイトボード」には、この実装通りで問題ないとおもいます。ユーザー側で、図解が欲しいと思った際に選択できる機能が良いと思いました。
<img width="686" height="492" alt="image" src="https://github.com/user-attachments/assets/a6dbed19-3d9a-4ae4-b675-e5e3ee027d18" />

daisuke-awaji committed:
feat(prompts): add summary and detail meeting styles, remove diagram instructions

- Add 'summary' style for concise meeting summaries with key points, decisions, and action items
- Add 'detail' style for comprehensive meeting records with full discussion flow
- Remove Mermaid diagram guidelines from newspaper, faq, and transcription prompts
- Simplify existing prompts by removing appended diagram instructions

These changes provide more specialized options for meeting transcription processing while streamlining prompts that don't require diagram visualization guidance.

Sugi275 reviewed (approved):
ご指摘修正ありがとうございます！ LGTM です！
